src/vtcomposite.cpp

```diff
@@ -826,104 +824,130 @@ struct LocalizeWorker : Napi::AsyncWorker
     }
 
-    if (change_names && language.length() == 0)
+    Napi::Object buffer_obj = buffer_val.As<Napi::Object>();
+    if (!buffer_obj.IsBuffer())
     {
-        return utils::CallbackError("language value is an empty string", info);
+        return utils::CallbackError("params.buffer is not a true Buffer", info);
     }
+    buffer = buffer_obj.As<Napi::Buffer<char>>();
 
-    // validate worldview string
-    Napi::Value worldview_val = info[2];
-    std::string worldview;
-
-    if (!worldview_val.IsString() && !worldview_val.IsNull())
+    // params.language (optional)
+    if (params.Has(Napi::String::New(info.Env(), "language")))
     {
-        return utils::CallbackError("worldview value must be null or a 2 character string", info);
+        Napi::Value language_val = params.Get(Napi::String::New(info.Env(), "language"));
+        if (!language_val.IsString() && !language_val.IsNull() && !language_val.IsUndefined())
+        {
+            return utils::CallbackError("params.language must be null or a string", info);
+        }
+        if (language_val.IsString())
+        {
+            language = language_val.As<Napi::String>();
+            if (language.length() == 0)
+            {
+                return utils::CallbackError("params.language cannot be an empty string", info);
+            }
+        }
+        if (language_val.IsNull() || language_val.IsUndefined())
+        {
+            change_names = false;
+        }
+    }
+    else
+    {
+        change_names = false;
     }
 
-    if (worldview_val.IsString())
+    // params.worldview
+    if (params.Has(Napi::String::New(info.Env(), "worldview")))
     {
-        worldview = worldview_val.As<Napi::String>();
-        if (worldview.length() != 2)
+        Napi::Value worldview_val = params.Get(Napi::String::New(info.Env(), "worldview"));
+        if (!worldview_val.IsArray())
         {
-            return utils::CallbackError("worldview must be a string 2 characters long", info);
+            return utils::CallbackError("params.worldview must be an array", info);
+        }
+        Napi::Array worldview_array = worldview_val.As<Napi::Array>();
+        std::uint32_t num_worldviews = worldview_array.Length();
+        worldview.reserve(num_worldviews);
+        for (std::uint32_t wv = 0; wv < num_worldviews; ++wv)
+        {
+            Napi::Value worldview_item_val = worldview_array.Get(wv);
+            if (!worldview_item_val.IsString())
+            {
+                return utils::CallbackError("params.worldview must be an array of strings", info);
+            }
+            std::string worldview_item = worldview_item_val.As<Napi::String>();
+            if (worldview_item.length() != 2)
+            {
+                return utils::CallbackError("params.worldview items must be strings of 2 characters", info);
+            }
+            worldview.emplace(worldview.end(), worldview_item);
```

Comment:
I don't think there's a reason to use emplace here... In fact, I think that may unnecessarily create a copy of the string. `push_back` would seem to be more appropriate

Reply:
+1